File: QuickApp/ClientApp/src/app/models/MoniorModels/phaseOut.ts
import { PersonalData } from "../app-add/personalData";
import { Monitor } from "./employeeMonitorModel";

export class phaseOutMonitor{
    public id: number;
    public fileNumber: string;
    public decisionMade: string;
    public situation: string;
    public monthAndYear: string;
    public planedBusiness: string;
    public reasonForPhaseOut: string;

    public entileMentFundReceive: string;
    public useOfEntileFund: string;
    public socialWellBeing: string;
    public isInterestOfFixedDeposit: boolean;
    public isLivesWithcloseFamily: boolean;

    public isIncomeGeneratingSources: boolean;
    public personID: number;
    public monitorPeriod: string;

    public summary:string;

    public monitorPeriodId:number;

    public economyWellBeign:string;

    public monitordate:Date

    public contactable:boolean;

    public isourintervention:boolean;

    public ourinterventionnote:string;

    public satisfactionlevel:string;

    public requirementnote:string;
    
    public betterphysicallivingcondition:boolean;

    public bettercaringandprotection:boolean;

    public healthy:boolean;


    public otherincomegeneratingnote:string;

    public dependonclosefamily:boolean

    public isaprequirement:boolean
    
    
    

    public monitorId: number;

    public person: PersonalData;

    public phaseOutMonitorImages : Array<PhaseOutMonitorImages> = new Array<PhaseOutMonitorImages>();

    public phaseOutMonitorDocuments : Array<PhaseOutMonitorImages> = new Array<PhaseOutMonitorImages>();

    public phaseOutMonitorVideos : Array<PhaseOutMonitorImages> = new Array<PhaseOutMonitorImages>();

    public monitor: Monitor = new Monitor();
}


export class PhaseOutMonitorImages{

    public id: number;

    public path: string;

    public fileName: string;

    public fileType: string;

}

File: DAL/Models/Monitoring Information/phaseOut.cs
using DAL.Models.Customer_Related;
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models.Monitoring_Information
{
  public class phaseOut
    {
        public int Id { get; set; }

        public string FileNumber { get; set; }

        public string DecisionMade { get; set; }

        public string Situation { get; set; }

        public string MonthAndYear { get; set; }

        public string PlanedBusiness { get; set; }

        public string ReasonForPhaseOut { get; set; }

        public string EntileMentFundReceive { get; set; }

        public string UseOfEntileFund { get; set; }

        public string SocialWellBeing { get; set; }

        public string  EconomyWellBeign { get; set; }

        public bool IsInterestOfFixedDeposit { get; set; }

        public bool IsLivesWithcloseFamily { get; set; }

        public bool IsIncomeGeneratingSources { get; set; }

        public int? PersonID { get; set; }

   
        public int monitorPeriodId { get; set; }

        public string  Summary { get; set; }

        public DateTime Monitordate { get; set; }

        public DateTime Dateofphaseout { get; set; }

        public bool Contactable { get; set; }

        public string Otherincomegeneratingnote { get; set; }

        public bool Dependonclosefamily { get; set; }

        public bool Betterphysicallivingcondition { get; set; }

        public bool Bettercaringandprotection { get; set; }

        public bool Healthy { get; set; }

        public bool Isaprequirement { get; set; }

        public string requirementnote { get; set; }

        public bool Isourintervention { get; set; }

        public string Satisfactionlevel { get; set; }

        public string Ourinterventionnote { get; set; }

        public Person person { get; set; }

     

        public ICollection<PhaseOutMonitorImages> phaseOutMonitorImages { get; set; }

        public ICollection<PhaseOutMonitorDocuments> phaseOutMonitorDocuments { get; set; }

        public ICollection<PhaseOutMonitorVideos> phaseOutMonitorVideos { get; set; }



        public MonitorPerioid monitorPeriod { get; set; }


    }
}
